src/mixins/ui/lasso-tool-ui.js

```diff
@@ -467,18 +659,10 @@ class PolylineShapeHandler extends ShapeHandler {
         )
       )
 
-      // NOTE: we're intentionally avoiding setting this.polyShape
```

Comment:
workarounds like what this comment describes is no longer necessary since we can control whether primary shapes should be destroyed or not in the `destroy` method.